docs/DafnyRef/Topics.md

```diff
@@ -34,6 +34,20 @@ const m := set x | n in {-1, 0, 1, 2} && f(n) >= 1
 const m := set x | n in {-1, 0, 1, 2} && g(n) >= 1
 ```
 
+A special case to be aware of: if the type in the comprehension differs from the collection,
+but both are compilable, the type of the comprehension will be fully tested, which could be viewed as a performance problem if only one specific trait needed to be tested. For example:
```

Comment:
`the type of the comprehension will be fully tested` => `at runtime, a filter is applied to the elements from the collection that filters out any values not of the comprehension type.`

Reply:
"A special case to be aware of: if the type in the comprehension differs from the collection,
but both are compilable, at runtime, every subset constraint yields a filter to be applied to the elements from the collection to keep only values of the comprehension type. Filtering starts either at the base type, or at the collection type if it happens to be a parent of the comprehension type.
In the first case, it is still possible to customize the emission of the filter to avoid testing redundant constraints. For example:"